### DISCUSSION TITLE
List with Domains (Regexp) working and not working

### DISCUSSION THREAD

**Discussion by cthulhubuddha** [evilsocket/opensnitch#877] [General]
ubuntu, opensnitch v1.6.0rc3

i am trying to limit the domains the application qemu can connect to. I have an always on rule using the domain regexp list
![image](https://user-images.githubusercontent.com/43973651/223719084-c5c1815e-a141-4033-a218-ab5932fa0cfb.png)

The domain list file has one entry:
^ntp.ubuntu.com$

The rule's To this port:
![image](https://user-images.githubusercontent.com/43973651/223720860-2275a926-8f24-4471-be52-f90cd4f1823d.png)

Log Level: Debug - the rule matches the regexp and then prompts me with an action popup
![image](https://user-images.githubusercontent.com/43973651/223720480-8859fb1c-7ad2-484c-afee-2ed7a3a6d699.png)

It appears my always on rule for this application matches the ntp.ubuntu.com DNS/53 call
![image](https://user-images.githubusercontent.com/43973651/223722987-ebb67335-854d-44e6-ad94-4b971abddf7a.png)

But doesn't match the ntp.ubuntu.com NTP/123 call, prompting the popup. When I add the temporary rule for ntp.ubuntu.com those ntp calls show there
![image](https://user-images.githubusercontent.com/43973651/223723489-1e73b740-f2d7-4613-b66b-fc7771229d36.png)

Not sure at this point if i'm doing something wrong or if there is a bug, looking for some ideas, thank you

**Comment by gustavo-iniguez-goya** [comment]
hi @cthulhubuddha ,

First of all upgrade to latest 1.6.0rc5. We've fixed several important bugs since rc3.

Then check `[x] Priority rule` for the allow rule. That will make opensnitch to allow that rule as soon as it matches a connection. Otherwise it'll continue evaluating rules until a deny rule matches or there're no more rules to evaluate.

**Reply by cthulhubuddha** [reply]
confirmed. I don't see an update option in the gui, the there a command line equivalent or just install the new .deb?

**Reply by gustavo-iniguez-goya** [reply]
download the debs from the release page: https://github.com/evilsocket/opensnitch/releases/tag/v1.6.0-rc.5

**Comment by cthulhubuddha** [comment]
OK, all upgraded (1.6.0rc5) and Priority Rule is selected, same behavior

**Reply by gustavo-iniguez-goya** [reply]
ok, check only the Regexp list checkbox, and deselect the ports and network ranges checkbox, to be sure that the list is working.

Take a look at this guide: https://github.com/evilsocket/opensnitch/wiki/block-lists#limiting-to-what-domains-an-application-can-connect-to

Also, restrict `systemd-resolved` to connect only to your DNS nameservers IP, port 53. 

By the way, is qemu a flatpak application or is it a system application (non containerized)?
and what process monitor method are you using (Preferences -> Nodes?

**Reply by cthulhubuddha** [reply]
I removed ports and network ranges, both 53 and 123 hit the base qemu rule (good). I then re-added ports, and both hit the base qemu rule again (good). When I added back the network ranges that's when it duplicated the above behavior - 53 to ntp.ubuntu.com hit the base but does not hit 123 and prompts me if I want to allow it (bad).

I added the actual IPs of ntp.ubuntu.com into the network range list but it still popped a query about ntp.ubuntu.com so it appears there is a conflict with the ranges?

Boxes is flatpak and process monitor is ebpf

**Reply by gustavo-iniguez-goya** [reply]
> I added the actual IPs of ntp.ubuntu.com into the network range list but it still popped a query about ntp.ubuntu.com so it appears there is a conflict with the ranges?

The problem is that in order a rule to match a connection, all the selected fields must match.
In this case, `ntp.ubuntu.com` is matching the regexp list, but probably the IP is not in the network ranges list.

Sometimes, chronyd, systemd-resolved, or others apps establish a connection against 127.0.0.1:53 or 127.0.0.53:53 (or 1.1.1.1, etc...). If those IPs are not in the network ranges list, then the rule won't match and will continue evaluating rules.

Take a look at the logs, and look for `new connection.*`

You can also double click on the Hosts tab over the ntp.ubuntu.com, there'll probably be connections to 127.0.0.1 or others DNS server IPs.

For example in my system:

![image](https://user-images.githubusercontent.com/2742953/224104819-d22eec00-26a4-48dc-b4f4-d29c1e769557.png)
